### PR TITLE
ZNTA-1657-TM-suggestion-modal

### PR DESCRIPTION
**Pull request by kathryngo**
JIRA issue URL: https://zanata.atlassian.net/browse/ZNTA-1657

- Fixed layout of TM Suggestions modal so that it can handle long file names for project, version and document. 
- Added titles for icons in modal
- Made sure the modal appears in front of the zanata logo (without changes to zanata-ui).
- Added Source and Target headings

tmdetails-modal

## Reviewer tasks

The following is based on the
[Zanata Development Guidelines](https://github.com/zanata/zanata-server/wiki/Development-Guidelines).
As a reviewer, you should check the guidelines regularly for updates, and just
to refresh your memory.

The reviewer can use this list for reference to make sure
they have considered all the important points.

Use the checkboxes or not, whatever works best for you.

 - Code quality
   - [ ] Code passes style check (checkstyle/eslint)
   - [ ] Code is clear and concise
   - [ ] UI code is internationalised
   - [ ] Code has adequate comments where appropriate
   - Code is in an appropriate place
     - [ ] Classes are in appropriate packages
     - [ ] Code fits with class's responsibilities (SRP)
   - [ ] Configuration files are documented enough
   - If code is removed
     - [ ] No remaining code references the removed code
       - [ ] No orphaned rewrite rules
       - [ ] No orphaned config settings
     - [ ] No remaining comments refer to the removed code
     - [ ] Unused imports and libraries are removed
 - Testing
   - [ ] New code has tests
   - [ ] Changed code has tests
   - [ ] All tests pass
   - [ ] Test coverage stable or increased
 - Documentation
   - [ ] New features are documented
   - [ ] Docs removed for deleted features
   - [ ] Docs updated for all changed features
   - [ ] Developer/sysadmin docs updated if appropriate
 - If there are new dependencies
   - [ ] Does each new dependency pass all the
         [Considerations for new dependencies](https://github.com/zanata/zanata-server/wiki/Development-Guidelines#new-technologies-and-dependencies)?

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

---
This change is [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/153)

### CODE REVIEW

**Comment by aeng** [comment]
Reviewed 2 of 5 files at r3.
Review status: 2 of 5 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[server/zanata-frontend/src/editor/app/containers/SuggestionDetailsModal/index.js, line 24 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/153#-KanUuiqxTYaDGUT9SSK:-KanUuirN8p5ItDwpoYg:b-kxtsxq) ([raw file](https://github.com/zanata/zanata-platform/blob/74caa45f3a3cc5016fbb71f94fa025655fc5c4e3/server/zanata-frontend/src/editor/app/containers/SuggestionDetailsModal/index.js#L24)):*
> ```JavaScript
>    * One of the detail panels is selected.
>    */
>   handleSelect (activeKey) {
> ```

Why remove this?

---

*[server/zanata-frontend/src/editor/app/containers/SuggestionDetailsModal/index.js, line 60 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/153#-KanUwGt1sNgxzQ50Bea:-KanUwGt1sNgxzQ50Beb:b4rrn3h) ([raw file](https://github.com/zanata/zanata-platform/blob/74caa45f3a3cc5016fbb71f94fa025655fc5c4e3/server/zanata-frontend/src/editor/app/containers/SuggestionDetailsModal/index.js#L60)):*
> ```JavaScript
>           <PanelGroup accordion
>             activeKey={this.state.activeKey}
>             onSelect={::this.handleSelect}>
> ```

Why onSelect is removed?

---

*[server/zanata-frontend/src/editor/app/containers/SuggestionDetailsModal/LocalProjectDetailPanel.js, line 77 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/153#-KanV6wKdjjAMX7hv0-l:-KanV6wKdjjAMX7hv0-m:bxzye5v) ([raw file](https://github.com/zanata/zanata-platform/blob/a1fc4d6a9567da9640626d08948b7118aa3e7509/server/zanata-frontend/src/editor/app/containers/SuggestionDetailsModal/LocalProjectDetailPanel.js#L77)):*
> ```JavaScript
>             <Row>
>               <Icon name="document" size="1" />DOCUMENT:
>               <span className="doc-name">{fullDocName}</span>
> ```

I'm quite worried about how we have css classes just for a label in a component. This does not encourage reusability and might introduce inconsistency on styling across all pages. e.g. `project-name, version-name and doc-name, transmem-slug`.

I prefer to have the refactoring done now than later.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/153)*

**Comment by kathryngo** [comment]
Review status: 2 of 5 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[server/zanata-frontend/src/editor/app/containers/SuggestionDetailsModal/index.js, line 24 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/153#-KanUuiqxTYaDGUT9SSK:-Kanmy2qPyIQQRanRU7e:b-8ek27u) ([raw file](https://github.com/zanata/zanata-platform/blob/74caa45f3a3cc5016fbb71f94fa025655fc5c4e3/server/zanata-frontend/src/editor/app/containers/SuggestionDetailsModal/index.js#L24)):*
<details><summary><i>Previously, aeng (Alex Eng) wrote…</i></summary><blockquote>

Why remove this?
</blockquote></details>

It was needed for the accordion which is no longer used

---

*[server/zanata-frontend/src/editor/app/containers/SuggestionDetailsModal/index.js, line 60 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/153#-KanUwGt1sNgxzQ50Bea:-Kann0jMlAdqlk-YZHbJ:b-8ek27u) ([raw file](https://github.com/zanata/zanata-platform/blob/74caa45f3a3cc5016fbb71f94fa025655fc5c4e3/server/zanata-frontend/src/editor/app/containers/SuggestionDetailsModal/index.js#L60)):*
<details><summary><i>Previously, aeng (Alex Eng) wrote…</i></summary><blockquote>

Why onSelect is removed?
</blockquote></details>

It was needed for the accordion which is no longer used

---

*[server/zanata-frontend/src/editor/app/containers/SuggestionDetailsModal/LocalProjectDetailPanel.js, line 77 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/153#-KanV6wKdjjAMX7hv0-l:-KanoDfmMQAeJ4e6Elp7:brvzvsc) ([raw file](https://github.com/zanata/zanata-platform/blob/a1fc4d6a9567da9640626d08948b7118aa3e7509/server/zanata-frontend/src/editor/app/containers/SuggestionDetailsModal/LocalProjectDetailPanel.js#L77)):*
<details><summary><i>Previously, aeng (Alex Eng) wrote…</i></summary><blockquote>

I'm quite worried about how we have css classes just for a label in a component. This does not encourage reusability and might introduce inconsistency on styling across all pages. e.g. `project-name, version-name and doc-name, transmem-slug`.

I prefer to have the refactoring done now than later.
</blockquote></details>

I don't know why I created seperate classes, apologies. There needs to be some styling applied but I can do it with one class named transUnit-label

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/153)*

**Comment by aeng** [comment]
:lgtm:

---

Reviewed 1 of 5 files at r3, 1 of 3 files at r4.
Review status: 3 of 5 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/153#-:-Kanx7w9F_Iyk3NSULiK:bnfp4nl)*

**Comment by aeng** [comment]
Reviewed 2 of 3 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/153)*

File: server/zanata-frontend/src/frontend/app/styles/style.less [hunk 1]
```diff
@@ -4677,6 +4698,20 @@ button.btn-link {
     min-height: 100%;
   }
 
+  main {
```
Comment (kathryngo) [inline server/zanata-frontend/src/frontend/app/styles/style.less]: for changes to dashboard and a few other pages

**Comment by spathare21** [comment]
@kathryngo @aeng it is still under review or can I  test it

**Comment by kathryngo** [comment]
@spathare21, I think @aeng should have a look because I made a few changes to the page layouts that I discussed with him earlier that are not mentioned in the PR

**Comment by aeng** [comment]
Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/153)*

**Comment by spathare21** [comment]
Verified

**Pull request merged**